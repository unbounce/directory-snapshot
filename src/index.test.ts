import { dss } from './index';

describe('directory', () => {
  test('creates a serializable directory', () => {
    expect(dss('./src/__tests__/project')).toMatchSnapshot();
  });
});
