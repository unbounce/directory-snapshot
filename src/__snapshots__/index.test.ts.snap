// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`directory creates a serializable directory 1`] = `
Object {
  "contents": Array [
    Object {
      "contents": "{
  \\"name\\": \\"project\\",
  \\"version\\": \\"1.0.0\\",
  \\"main\\": \\"index.js\\",
  \\"license\\": \\"MIT\\"
}
",
      "path": "package.json",
      "stats": Object {
        "isDirectory": false,
        "isFile": true,
        "isSocket": false,
        "isSymbolicLink": false,
      },
    },
    Object {
      "contents": "",
      "path": "src",
      "stats": Object {
        "isDirectory": true,
        "isFile": false,
        "isSocket": false,
        "isSymbolicLink": false,
      },
    },
    Object {
      "contents": "console.log('hello, world!');
",
      "path": "src/index.js",
      "stats": Object {
        "isDirectory": false,
        "isFile": true,
        "isSocket": false,
        "isSymbolicLink": false,
      },
    },
  ],
}
`;
